src/plonkish/ir/sc.rs

```diff
@@ -114,3 +124,120 @@ impl<F: Field + Hash, MappingArgs> MappingGenerator<F, MappingArgs> {
         ctx.get_super_assignments()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use halo2curves::bn256::Fr;
+
+    use super::*;
+
+    #[test]
+    fn test_default() {
+        let super_circuit: SuperCircuit<Fr, ()> = Default::default();
+
+        assert_eq!(
+            format!("{:#?}", super_circuit.sub_circuits),
+            format!("{:#?}", Vec::<Circuit<Fr>>::default())
+        );
+        assert_eq!(
+            format!("{:#?}", super_circuit.mapping),
+            format!("{:#?}", MappingGenerator::<Fr, ()>::default())
+        );
+    }
+
+    #[test]
+    fn test_add_sub_circuit() {
+        let mut super_circuit: SuperCircuit<Fr, ()> = Default::default();
+
+        let sub_circuit = Circuit::<Fr>::default();
+
+        super_circuit.add_sub_circuit(sub_circuit.clone());
+
+        assert_eq!(super_circuit.sub_circuits.len(), 1);
+        assert_eq!(
+            format!("{:#?}", super_circuit.sub_circuits[0]),
+            format!("{:#?}", sub_circuit)
+        );
+    }
+
+    #[test]
+    fn test_get_mapping() {
+        let super_circuit: SuperCircuit<Fr, ()> = Default::default();
+
+        let mapping = super_circuit.get_mapping();
+
+        assert_eq!(
+            format!("{:#?}", mapping),
+            format!("{:#?}", MappingGenerator::<Fr, ()>::default())
+        );
+    }
+
+    #[test]
+    fn test_set_mapping() {
+        let mut super_circuit: SuperCircuit<Fr, ()> = Default::default();
+
+        super_circuit.set_mapping(|ctx, args| {
+            // Your mapping logic here
+            ctx.map(&AssignmentGenerator::<Fr, ()>::default(), args)
+        });
+
+        assert_eq!(
+            format!("{:#?}", super_circuit.mapping),
+            format!("{:#?}", MappingGenerator::<Fr, ()>::default())
+        );
+    }
+
+    #[test]
+    fn test_get_sub_circuits() {
+        let super_circuit: SuperCircuit<Fr, ()> = Default::default();
+
+        let sub_circuits = super_circuit.get_sub_circuits();
+
+        assert_eq!(
+            format!("{:#?}", sub_circuits),
+            format!("{:#?}", Vec::<Circuit<Fr>>::default())
+        );
+    }
+
+    #[test]
+    fn test_mapping_context_default() {
+        let ctx = MappingContext::<Fr>::default();
+
+        assert_eq!(
+            format!("{:#?}", ctx.assignments),
+            format!("{:#?}", SuperAssignments::<Fr>::default())
+        );
+    }
+
+    #[test]
+    fn test_mapping_context_map() {
+        let mut ctx = MappingContext::<Fr>::default();
```

Comment:
@rutefig there are many tests in this file that the defaults are defaults, that I think it is not interesting. What about actually putting some data and doing some tests about the data processing